src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2279,6 +2336,33 @@ def _reindex_axis_0(
                 modin_frame.data_column_snowflake_quoted_identifiers
             )
         )
+        if is_index:
+            # We want to remove the monotonic_increasing and monotonic_decreasing columns so that they will not be filled here.
```

Comment:
I'm confused about what the internal frame we're creating here is used for if it doesn't have the `monotonic_increasing` or `monotonic_decreasing` columns. Could you add a comment explaining what it's used for and how it relates to `new_qc`?

Reply:
We need the monotonic increasing and decreasing columns physically in the data in order to use it later on, but we don't want them filled by the fillna column. To keep them around after the join, they need to be in the data columns for the internal frame though, as the join purges non active columns from the current view of the snowpark dataframe (still physically present, but no longer accessible from the current view). We basically just need them to check at the very end if the original data was monotonic increasing or decreasing. Will go ahead and add this explanation to the comment as well!

Reply:
Okay, I think the comments help clarify what is happening now.